flutter/lib/src/navigation/display_strategy_evaluator.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+@internal
+class DisplayStrategyEvaluator {
+  static final DisplayStrategyEvaluator _instance =
+      DisplayStrategyEvaluator._internal();
+
+  factory DisplayStrategyEvaluator() {
+    return _instance;
+  }
+
+  DisplayStrategyEvaluator._internal();
+
+  final Map<String, bool> _manualReportReceived = {};
+  final Map<String, Timer> _timers = {};
+  final Map<String, Completer<TimeToDisplayStrategy>> _completers = {};
```

Comment:
`m`: If I'm not mistaken, the key for each map is the `route`. I think you could merge these three maps into one. You get two major benefits: you don't have to manage the state of three different maps, which then is easy to get wrong. If you want to remove a route, you have to remove it from all three maps. The second benefit is that you use less memory.